examples/Custom_Hooks/campaign_stopping.py

```diff
@@ -0,0 +1,238 @@
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace, SearchSpaceType
+from baybe.simulation import simulate_experiment, simulate_scenarios
+from baybe.targets import NumericalTarget
+from baybe.utils import register_hooks
+from baybe.utils.dataframe import to_tensor
+from baybe.utils.plotting import create_example_plots
+from baybe.utils.random import set_random_seed
+
+### Temporary
+warnings.filterwarnings(
+    "ignore", category=UnusedObjectWarning, message="explicit objective"
+)
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+
+### Settings
+
+# For the full simulation, we need to define some additional settings.
+# These are the number of Monte Carlo runs, the number of experiments to be conducted
+# per run and number of campaigns we run with the stopping criterion.
+# We also fix the random seed for reproducibility.
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+
+N_DOE_ITERATIONS = 2 if SMOKE_TEST else 20
+N_MC_ITERATIONS = 2 if SMOKE_TEST else 50
+N_STOPPED_CAMPAIGNS = 2 if SMOKE_TEST else 5
+BATCH_SIZE = 1
+set_random_seed(1337)
+
+### Lookup Functionality and Parameter Data
+
+# This loads the dataframe used for looking up measurements in the simulation.
+
+try:
+    lookup = pd.read_excel("./../Backtesting/lookup.xlsx")
+except FileNotFoundError:
+    try:
+        lookup = pd.read_excel("examples/Backtesting/lookup.xlsx")
+    except FileNotFoundError as e:
+        print(e)
+
+# Then we define the substance categories and SMILES representing their molecular structure.
+# Note that we need to ensure that the names fit the names in the provided .xlsx file!
+
+dict_solvent = {
+    "DMAc": r"CC(N(C)C)=O",
+    "Butyornitrile": r"CCCC#N",
+    "Butyl Ester": r"CCCCOC(C)=O",
+    "p-Xylene": r"CC1=CC=C(C)C=C1",
+}
+dict_base = {
+    "Potassium acetate": r"O=C([O-])C.[K+]",
+    "Potassium pivalate": r"O=C([O-])C(C)(C)C.[K+]",
+    "Cesium acetate": r"O=C([O-])C.[Cs+]",
+    "Cesium pivalate": r"O=C([O-])C(C)(C)C.[Cs+]",
+}
+dict_ligand = {
+    "BrettPhos": r"CC(C)C1=CC(C(C)C)=C(C(C(C)C)=C1)C2=C(P(C3CCCCC3)C4CCCCC4)C(OC)="
+    "CC=C2OC",
+    "Di-tert-butylphenylphosphine": r"CC(C)(C)P(C1=CC=CC=C1)C(C)(C)C",
+    "(t-Bu)PhCPhos": r"CN(C)C1=CC=CC(N(C)C)=C1C2=CC=CC=C2P(C(C)(C)C)C3=CC=CC=C3",
+    "Tricyclohexylphosphine": r"P(C1CCCCC1)(C2CCCCC2)C3CCCCC3",
+    "PPh3": r"P(C1=CC=CC=C1)(C2=CC=CC=C2)C3=CC=CC=C3",
+    "XPhos": r"CC(C1=C(C2=CC=CC=C2P(C3CCCCC3)C4CCCCC4)C(C(C)C)=CC(C(C)C)=C1)C",
+    "P(2-furyl)3": r"P(C1=CC=CO1)(C2=CC=CO2)C3=CC=CO3",
+    "Methyldiphenylphosphine": r"CP(C1=CC=CC=C1)C2=CC=CC=C2",
+    "1268824-69-6": r"CC(OC1=C(P(C2CCCCC2)C3CCCCC3)C(OC(C)C)=CC=C1)C",
+    "JackiePhos": r"FC(F)(F)C1=CC(P(C2=C(C3=C(C(C)C)C=C(C(C)C)C=C3C(C)C)C(OC)=CC=C2OC)"
+    r"C4=CC(C(F)(F)F)=CC(C(F)(F)F)=C4)=CC(C(F)(F)F)=C1",
+    "SCHEMBL15068049": r"C[C@]1(O2)O[C@](C[C@]2(C)P3C4=CC=CC=C4)(C)O[C@]3(C)C1",
+    "Me2PPh": r"CP(C)C1=CC=CC=C1",
+}
+
+### Run the Scenario for the Unstopped Campaign
```

Comment:
`Unstopped` sounds weird. Rather `full` or `uninterrupted`?

Reply:
changed to uninterrupted